packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -263,27 +351,34 @@ function* getNodeDiscrepancies(
 }
 
 /**
- * The function to track the discrepancies between two field stored schemas.
+ * The function to track the discrepancies between a field view schema and a stored schema.
  *
  * @param keyOrRoot - If the key is missing, it indicates that this is the root field schema.
  */
 function* getFieldDiscrepancies(
-	view: TreeFieldStoredSchema,
+	view: FieldSchema,
 	stored: TreeFieldStoredSchema,
 	identifier: TreeNodeSchemaIdentifier | undefined,
 	fieldKey: FieldKey | undefined,
+	viewKindIsSequence = false,
 ): Iterable<FieldDiscrepancy> {
+	const normalizedView = normalizeFieldSchema(view);
+
 	// Only track the symmetric differences of two sets.
 	const findSetDiscrepancies = (
-		a: TreeTypeSet,
+		a: readonly AnnotatedAllowedType<TreeNodeSchema>[],
 		b: TreeTypeSet,
-	): [TreeNodeSchemaIdentifier[], TreeNodeSchemaIdentifier[]] => {
-		const aDiff = [...a].filter((value) => !b.has(value));
-		const bDiff = [...b].filter((value) => !a.has(value));
+	): [readonly AnnotatedAllowedType<TreeNodeSchema>[], TreeNodeSchemaIdentifier[]] => {
+		const aIdentifiers = new Set(a.map((value) => value.type.identifier));
```

Comment:
I'd pick a more descriptive name given that our view schema system has a special notion of identifier fields, so this could be confusing.

`aNodeSchemaIdentifiers` would be better.

Also would be good to rename a and b now that this is no longer symmetric.